tests/ignite/contrib/test_engines.py

```diff
@@ -83,11 +80,59 @@ def _test_create_supervised_tbptt_trainer(device):
     # Verifications
     assert handle_started.call_count == 3
     assert handle_completed.call_count == 3
+    assert mock_detach_hidden.call_count == 2
+    assert model.call_count == 3
+    assert loss.call_count == 3
+    assert optimizer.zero_grad.call_count == 3
+    assert optimizer.step.call_count == 3
+    n_args_tuple = tuple(len(args) for args, kwargs in model.call_args_list)
+    assert n_args_tuple == (1, 2, 2)
+
+
+def _test_create_supervised_tbptt_trainer(device):
+    # Defining dummy recurrent model with zero weights
+    model = nn.RNN(1, 1, bias=False)
+    for p in model.parameters():
+        p.data.zero_()
+
+    # Set some mock on forward to monitor
+    forward_mock = mock.MagicMock()
+    forward_mock.return_value = None
+    model.register_forward_hook(forward_mock)
+
+    # Defning optimizer and trainer
+    optimizer = optim.SGD(model.parameters(), 1)
+    trainer = create_supervised_tbptt_trainer(
+        model,
+        optimizer,
+        F.mse_loss,
+        tbtt_step=2,
+        device=device
+    )
+
+    # Fake data
+    X = torch.ones(6, 2, 1)
+    y = torch.ones(6, 2, 1)
+    data = [(X, y)]
+
+    # Running trainer
+    trainer.run(data)
 
     # If tbptt is not use (one gradient update), the hidden to hidden weight
     # should stay zero
     assert not model.weight_hh_l0.item() == pytest.approx(0)
 
+    # Cheking forward calls
+    assert forward_mock.call_count == 3
+    for i in range(3):
+        inputs = forward_mock.call_args_list[i][0][1]
+        if i == 0:
+            assert len(inputs) == 1
+        else:
+            assert len(inputs) == 2
```

Comment:
@vfdev-5 Here the number of arguments call to the model is 2 for input and hidden